src/library_sockfs.js

```diff
@@ -10,37 +10,35 @@ addToLibrary({
   },
   $SOCKFS__deps: ['$FS'],
   $SOCKFS: {
+#if expectToReceiveOnModule('websocket')
+    websocketArgs: {},
+#endif
+    callbacks: {},
+    on(event, callback) {
+      SOCKFS.callbacks[event] = callback;
+    },
+    emit(event, param) {
+      SOCKFS.callbacks[event]?.(param);
+    },
     mount(mount) {
-      // If Module['websocket'] has already been defined (e.g. for configuring
-      // the subprotocol/url) use that, if not initialise it to a new object.
-      Module['websocket'] = (Module['websocket'] &&
-                             ('object' === typeof Module['websocket'])) ? Module['websocket'] : {};
-
+#if expectToReceiveOnModule('websocket')
+      // The incomming Module['websocket'] can be used for configuring 
+      // configuring subprotocol/url, etc
+      SOCKFS.websocketArgs = {{{ makeModuleReceiveExpr('websocket', '{}') }}};
       // Add the Event registration mechanism to the exported websocket configuration
       // object so we can register network callbacks from native JavaScript too.
       // For more documentation see system/include/emscripten/emscripten.h
-      Module['websocket']._callbacks = {};
-      Module['websocket']['on'] = /** @this{Object} */ function(event, callback) {
-        if ('function' === typeof callback) {
-          this._callbacks[event] = callback;
-        }
-        return this;
-      };
-
-      Module['websocket'].emit = /** @this{Object} */ function(event, param) {
-        if ('function' === typeof this._callbacks[event]) {
-          this._callbacks[event].call(this, param);
-        }
-      };
+      (Module['websocket'] ??= {})['on'] = SOCKFS.on;
```

Comment:
Do we not need `'emit'` as well? Looks like before we had both exported.

Reply:
No, I don't think so.   The `emit` for for internal use to send the message.  The `on` is what users would use to register callback.

Reply:
You can see that the `on` property is quoted so that closure won't minify it but the `emit` helper function is not.

Reply:
See the docs here: https://emscripten.org/docs/api_reference/emscripten.h.html#socket-event-registration